seatunnel-connectors-v2/connector-typesense/src/main/java/org/apache/seatunnel/connectors/seatunnel/typesense/catalog/TypesenseTypeConverter.java

```diff
@@ -0,0 +1,94 @@
+import org.apache.seatunnel.api.table.converter.TypeConverter;
+import org.apache.seatunnel.api.table.type.BasicType;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.seatunnel.typesense.client.TypesenseType;
+
+import com.google.auto.service.AutoService;
+
+import java.util.Map;
+
+import static org.apache.seatunnel.connectors.seatunnel.typesense.client.TypesenseType.INT32;
+import static org.apache.seatunnel.connectors.seatunnel.typesense.client.TypesenseType.INT64;
+
+@AutoService(TypeConverter.class)
+public class TypesenseTypeConverter implements BasicTypeConverter<BasicTypeDefine<TypesenseType>> {
+    public static final TypesenseTypeConverter INSTANCE = new TypesenseTypeConverter();
+
+    @Override
+    public String identifier() {
+        return "Typesense";
+    }
+
+    @Override
+    public Column convert(BasicTypeDefine<TypesenseType> typeDefine) {
+        PhysicalColumn.PhysicalColumnBuilder builder =
+                PhysicalColumn.builder()
+                        .name(typeDefine.getName())
+                        .sourceType(typeDefine.getColumnType())
+                        .nullable(typeDefine.isNullable())
+                        .defaultValue(typeDefine.getDefaultValue())
+                        .comment(typeDefine.getComment());
+        String type = typeDefine.getDataType().toLowerCase();
+        switch (type) {
+            case INT32:
+                builder.dataType(BasicType.INT_TYPE);
+                break;
+            case INT64:
+                builder.dataType(BasicType.LONG_TYPE);
+                break;
+            case TypesenseType.FLOAT:
+                builder.dataType(BasicType.FLOAT_TYPE);
+                break;
+            case TypesenseType.BOOL:
+                builder.dataType(BasicType.BOOLEAN_TYPE);
+                break;
+            case TypesenseType.OBJET:
+                Map<String, BasicTypeDefine<TypesenseType>> typeInfo =
+                        (Map) typeDefine.getNativeType().getOptions();
+                SeaTunnelRowType object =
+                        new SeaTunnelRowType(
+                                typeInfo.keySet().toArray(new String[0]),
+                                typeInfo.values().stream()
+                                        .map(this::convert)
+                                        .map(Column::getDataType)
+                                        .toArray(SeaTunnelDataType<?>[]::new));
+                builder.dataType(object);
+                break;
+            case TypesenseType.STRING:
+            case TypesenseType.IMAGE:
+            default:
+                builder.dataType(BasicType.STRING_TYPE);
+                break;
+        }
+        return builder.build();
+    }
+
+    @Override
+    public BasicTypeDefine<TypesenseType> reconvert(Column column) {
+        return null;
```

Comment:
null ?

Reply:
@Hisoka-X 
This method is currently not used in the code. Do I still need to write in this situation? Do you have any good suggestions.

Reply:
throw new UnsupportedException?

Reply:
modify after:
```java
@Override
    public BasicTypeDefine<TypesenseType> reconvert(Column column) {
        throw new UnsupportedOperationException("Unsupported operation");
    }
```